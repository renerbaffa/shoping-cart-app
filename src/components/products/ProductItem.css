.ProductItem-container {
  width: 100%;
  display: flex;
  border-top: 1px solid lightgray;
  border-left: 1px solid lightgray;
  border-right: 1px solid lightgray;
}

.ProductItem-container-odd {
  background-color: #f2f2f2;
}

.ProductItem-image {
  height: 50px;
  width: calc(12% - 6px);
  margin: 3px;
}

.ProductItem-description-container {
  padding: 2px 12px;
  width: 60%;
}

.ProductItem-description-container2 {
  padding: 0px 12px;
  width: 20%;
  align-self: center;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
}

.ProductItem-name {
  font-weight: bold;
  font-size: 14px;
}

.ProductItem-description {
  font-size: 13px;
}

.ProductItem-bottom {
  width: 8%;
  align-self: center;
  display: flex;
  justify-content: center;
}

.ProductItem-button {
  background-color: #FF6D3E;
  padding: 1px 4px;
  border-radius: 50%;
  color: white;
  cursor: pointer;
}
