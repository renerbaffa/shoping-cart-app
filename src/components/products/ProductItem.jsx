import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import withProduct from '../../HOCs/withProduct';

import Image from '../shared/Image';
import Button from '../shared/Button';

import './ProductItem.css';

export const ProductItem = ({
  className,
  description,
  image,
  index,
  name,
  onAddProduct,
  unitPrice,
  unitsInStock,
  ...other
}) => (
  <div className={cx(
    className,
    'ProductItem-container',
    { 'ProductItem-container-odd' : index % 2 }
  )}>
    <Image
      className="ProductItem-image"
      alt={name}
      src={image}
    />
    <div className="ProductItem-description-container">
      <div className="ProductItem-name">{name}</div>
      <div className="ProductItem-description">{description}</div>
    </div>
    <div className="ProductItem-description-container2">
      <div>Price: <b>${unitPrice}</b></div>
      <div><b>{unitsInStock}</b> in stock</div>
    </div>
    <div className="ProductItem-bottom">
      <Button
        className="ProductItem-button"
        onClick={onAddProduct}
      >
        +
      </Button>
    </div>
  </div>
);

ProductItem.propTypes = {
  className: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  index: PropTypes.string,
  name: PropTypes.string,
  onAddProduct: PropTypes.func,
  unitPrice: PropTypes.number,
  unitsInStock: PropTypes.number,
}

ProductItem.defaultProps = {
  className: '',
  description: '',
  image: '',
  index: '',
  name: '',
  onAddProduct: () => {},
  unitPrice: 0,
  unitsInStock: 0,
}

export default withProduct(ProductItem);
