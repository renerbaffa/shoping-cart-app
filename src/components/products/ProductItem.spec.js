import React from 'react';
import renderer from 'react-test-renderer';

import { ProductItem } from './ProductItem';

describe('<ProductItem />', () => {
  it('should render correctly', () => {
    const productItem = renderer.create(
      <ProductItem
        className="className"
        description="description"
        image="image"
        index={0}
        name="Product Name"
        onAddProduct={() => {}}
        unitPrice={29.30}
        unitsInStock={12}
      />
    ).toJSON();
    expect(productItem).toMatchSnapshot();
  });

  it('should reach 100% of coverage', () => {
    ProductItem.defaultProps.onAddProduct();
  });
});
