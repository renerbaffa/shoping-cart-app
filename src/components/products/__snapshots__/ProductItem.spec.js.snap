// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem /> should render correctly 1`] = `
<div
  className="className ProductItem-container"
>
  <img
    alt="Product Name"
    className="ProductItem-image"
    src="image"
  />
  <div
    className="ProductItem-description-container"
  >
    <div
      className="ProductItem-name"
    >
      Product Name
    </div>
    <div
      className="ProductItem-description"
    >
      description
    </div>
  </div>
  <div
    className="ProductItem-description-container2"
  >
    <div>
      Price: 
      <b>
        $
        29.3
      </b>
    </div>
    <div>
      <b>
        12
      </b>
       in stock
    </div>
  </div>
  <div
    className="ProductItem-bottom"
  >
    <button
      className="ProductItem-button Button"
      onClick={[Function]}
    >
      +
    </button>
  </div>
</div>
`;
